app/components/paperShow/refCitedTab/index.tsx

```diff
@@ -24,37 +26,57 @@ const TabItem: React.FunctionComponent<TabItemProps> = props => {
 
 const PDFButton: React.FunctionComponent<PDFButtonProps> = props => {
   const [isOpen, setIsOpen] = React.useState(false);
+  const [isOpenBlockedPopper, setIsOpenBlockedPopper] = React.useState(false);
+  const actionBtnEl = React.useRef<HTMLDivElement | null>(null);
 
   if (props.canShowFullPDF) {
     return (
-      <div className={styles.actionItem}>
-        <PdfDownloadButton
-          paper={props.paper}
-          isLoading={props.isLoading}
-          onDownloadedPDF={props.onClickDownloadPDF!}
-          handleSetScrollAfterDownload={props.afterDownloadPDF}
-        />
-      </div>
+      <ClickAwayListener onClickAway={() => setIsOpenBlockedPopper(false)}>
+        <div className={styles.actionItem} ref={actionBtnEl}>
+          <PdfDownloadButton
+            paper={props.paper}
+            isLoading={props.isLoading}
+            isOpenBlockedPopper={isOpenBlockedPopper}
+            onDownloadedPDF={props.onClickDownloadPDF!}
+            handleSetIsOpenBlockedPopper={setIsOpenBlockedPopper}
+            handleSetScrollAfterDownload={props.afterDownloadPDF}
+          />
+          <BlockedPopper
+            handleOnClickAwayFunc={() => setIsOpenBlockedPopper(false)}
+            anchorEl={actionBtnEl.current}
+            isOpen={isOpenBlockedPopper}
+            buttonClickAction={"downloadPdf"}
+          />
+        </div>
+      </ClickAwayListener>
     );
   }
   return (
-    <>
-      <div className={styles.actionItem}>
+    <ClickAwayListener onClickAway={() => setIsOpenBlockedPopper(false)}>
```

Comment:
I think it's too duplicated code. the only different thing here is the content of the button.